doc/api/querystring.md

```diff
@@ -4,81 +4,140 @@
 
 <!--name=querystring-->
 
-This module provides utilities for dealing with query strings.
-It provides the following methods:
+The `querystring` module provides utilities for parsing and formatting URL
+query strings. It can be accessed using:
 
-## querystring.escape
+```js
+const querystring = require('querystring');
+```
+
+## querystring.escape(str)
 <!-- YAML
 added: v0.1.25
 -->
 
-The escape function used by `querystring.stringify`,
-provided so that it could be overridden if necessary.
+* `str` {String}
+
+The `querystring.escape()` method performs URL percent-encoding on the given
+`str` in a manner that is optimized for the specific requirements of URL
+query strings.
+
+The `querystring.escape()` method is used by `querystring.stringify()` and is
+generally not expected to be used directly. It is exported primarily to allow
+application code to provide a replacement percent-encoding implementation if
+necessary by assigning `querystring.escape` to an alternative function.
 
 ## querystring.parse(str[, sep][, eq][, options])
 <!-- YAML
 added: v0.1.25
 -->
 
-Deserialize a query string to an object.
-Optionally override the default separator (`'&'`) and assignment (`'='`)
-characters.
+* `str` {String} The URL query string to parse
+* `sep` {String} The substring used to delimit key+value pairs in the
+  query string. Defaults to `'&'`.
+* `eq` {String}. The substring used to delimit keys and values in the
+  query string. Defaults to `'='`.
+* `options` {Object}
+  * `decodeURIComponent` {Function} The function to use when decoding
```

Comment:
`unescape` is actually a JS function too, but with different behavior than `decodeURIComponent`. Changing it to that would probably make things even more confusing.